doc/source/notebooks/theory/Sanity_check.ipynb

```diff
@@ -134,164 +105,19 @@
-     ]
-    },
-    {
-     "name": "stderr",
-     "output_type": "stream",
-     "text": [
-      "INFO:tensorflow:Optimization terminated with:\n",
-      "  Message: Optimization terminated successfully.\n",
-      "  Objective function value: 18.834400\n",
-      "  Number of iterations: 52\n",
-      "  Number of functions evaluations: 70\n"
-     ]
-    },
-    {
-     "name": "stdout",
-     "output_type": "stream",
-     "text": [
-      "INFO:tensorflow:Optimization terminated with:\n",
-      "  Message: Optimization terminated successfully.\n",
-      "  Objective function value: 18.834400\n",
-      "  Number of iterations: 52\n",
-      "  Number of functions evaluations: 70\n"
-     ]
-    },
-    {
-     "name": "stderr",
-     "output_type": "stream",
-     "text": [
-      "INFO:tensorflow:Optimization terminated with:\n",
-      "  Message: Optimization terminated successfully.\n",
-      "  Objective function value: 18.834400\n",
-      "  Number of iterations: 52\n",
-      "  Number of functions evaluations: 70\n"
-     ]
-    },
-    {
-     "name": "stdout",
-     "output_type": "stream",
-     "text": [
-      "INFO:tensorflow:Optimization terminated with:\n",
-      "  Message: Optimization terminated successfully.\n",
-      "  Objective function value: 18.834400\n",
-      "  Number of iterations: 7\n",
-      "  Number of functions evaluations: 12\n"
-     ]
-    },
-    {
-     "name": "stderr",
-     "output_type": "stream",
-     "text": [
-      "INFO:tensorflow:Optimization terminated with:\n",
-      "  Message: Optimization terminated successfully.\n",
-      "  Objective function value: 18.834400\n",
-      "  Number of iterations: 7\n",
-      "  Number of functions evaluations: 12\n"
-     ]
-    },
-    {
-     "name": "stdout",
-     "output_type": "stream",
-     "text": [
-      "INFO:tensorflow:Optimization terminated with:\n",
-      "  Message: Optimization terminated successfully.\n",
-      "  Objective function value: 18.834355\n",
-      "  Number of iterations: 7\n",
-      "  Number of functions evaluations: 12\n"
-     ]
-    },
-    {
-     "name": "stderr",
-     "output_type": "stream",
-     "text": [
-      "INFO:tensorflow:Optimization terminated with:\n",
-      "  Message: Optimization terminated successfully.\n",
-      "  Objective function value: 18.834355\n",
-      "  Number of iterations: 7\n",
-      "  Number of functions evaluations: 12\n"
-     ]
-    }
-   ],
+   "outputs": [],
    "source": [
-    "_ = [gpflow.train.ScipyOptimizer(method='BFGS').minimize(m, maxiter=100) for m in models]"
+    "for m in models:\n",
+    "    opt = gpflow.optimizers.Scipy()\n",
+    "    if isinstance(m, gpflow.models.SVGP):\n",
+    "        loss_fn = lambda: m.neg_log_marginal_likelihood(data[0], data[1])\n",
+    "    else:\n",
+    "        loss_fn = m.neg_log_marginal_likelihood\n",
+    "    opt.minimize(loss_fn, variables=m.trainable_variables, options=dict(maxiter=1000))"
```

Comment:
Should this use `BFGS` (instead of the `L-BFGS-B` default) as well?

Is the optimization fast enough not to affect notebook test time? Or would it be worthwhile replacing the maxiter=1000 with `maxiter=ci_niter(1000)`?

Reply:
I will update maxiter with `ci_niter` and `L-BFGS-B` works fine.

Reply:
done